flytekit/core/type_engine.py

```diff
@@ -401,14 +403,22 @@ def get_transformer(cls, python_type: Type) -> TypeTransformer[T]:
             TODO lets make this deterministic by using an ordered dict
 
         """
+
         # Step 1
         if python_type in cls._REGISTRY:
             return cls._REGISTRY[python_type]
 
         # Step 2
         if hasattr(python_type, "__origin__"):
+
+            # Handling of annotated generics, eg:
+            # typing.Annotated[typing.List[int], 'foo']
+            if isinstance(python_type, typing._AnnotatedAlias):
+                return cls.get_transformer(python_type.__origin__)
```

Comment:
I know we've been using the `__origin__` dunder attribute in a few places, but let's take this opportunity to use [get_args](https://docs.python.org/3/library/typing.html#typing.get_args) to access the annotations.

Reply:
same here, scanning the documentation its unclear what is more "correct" and both do the same thing, so would like to understand why ( for my own understanding )

Reply:
This one I'm not too particularly passionate about since we use in a lot of other places already.

Reply:
will change because its dunder and didn't know those could break